mono/metadata/cominterop.c

```diff
@@ -378,30 +378,33 @@ cominterop_get_method_interface (MonoMethod* method)
 		}
 	}
 
-	if (!ic) 
-		g_assert (ic);
-	g_assert (MONO_CLASS_IS_INTERFACE (ic));
-
 	return ic;
 }
 
 /**
  * cominterop_get_com_slot_for_method:
  * @method: a method
+ * @error: set on error
  *
  * Returns: the method's slot in the COM interface vtable
  */
 static int
-cominterop_get_com_slot_for_method (MonoMethod* method)
+cominterop_get_com_slot_for_method (MonoMethod* method, MonoError* error)
 {
 	guint32 slot = method->slot;
  	MonoClass *ic = method->klass;
 
+	error_init (error);
+
 	/* if method is on a class, we need to look up interface method exists on */
 	if (!MONO_CLASS_IS_INTERFACE(ic)) {
 		int offset = 0;
 		int i = 0;
 		ic = cominterop_get_method_interface (method);
+		if (!ic || !MONO_CLASS_IS_INTERFACE (ic)) {
+			mono_error_set_invalid_program (error, "COM interface for method %s cannot be found.", mono_method_full_name (method, TRUE));
+			return -1;
```

Comment:
This will leak the result of mono_method_full_name (), you should do:
char *name = mono_method_full_name ();
set_invalid_program (...)
g_free (name);

Reply:
Good catch! Fixed and updated.